develop/api/CommunicationGatewayErrorCodes.md

```diff
@@ -0,0 +1,84 @@
+|DataLoss|Data loss|15|Unrecoverable data loss or corruption.|
+|Unauthenticated|Unauthenticated|16|The request does not have valid authentication credentials for the operation.|
+
+## Channel management handler error codes
+
+These are the error codes that can be returned when creating/connecting or disconnecting a gNMI handler in the CommunicationGateway.
+
+|Name|Friendly Name|Code|Message|
+|--- |--- |--- |--- |
+|ChannelManagementHandler_ConnectRequest_InvalidChannelInstance|Channel management ConnectRequest has invalid instance GUID|1001001|The ConnectRequest did not provide a usable channel instance GUID, which is needed as a message broker subject token. This should be a hex-string GUID without separators.|
+|ChannelManagementHandler_ConnectRequest_CreateChannelFailed|Channel management ConnectRequest's CreateChannel failed|1001002|The ConnectRequest failed to create a channel. Check the CommunicationGateway logging for more detailed error information. There's likely a problem with the provided channel configuration.|
+|ChannelManagementHandler_ConnectRequest_FailedToComplete|Channel management ConnectRequest failed to complete|1001003|The ConnectRequest failed to complete when creating the gNMI handler. Check the CommunicationGateway logging for more detailed error information.|
+|ChannelManagementHandler_DisconnectRequest_FailedToComplete|Channel management DisconnectRequest failed to complete|1001004|The DisconnectRequest failed to complete when releasing the channel and cleaning up resources. Check the CommunicationGateway logging for more detailed error information.|
+|ChannelManagementHandler_ConnectRequest_InvalidMessage|Channel management invalid ConnectRequest|1001005|The ConnectRequest could not be parsed. This topic should send a Request message containing a ConnectRequest.|
+|ChannelManagementHandler_DisconnectRequest_InvalidMessage|Channel management invalid DisconnectRequest|1001006|The DisconnectRequest could not be parsed. This topic should send a Request message containing a DisconnectRequest.|
+|ChannelManagementHandler_ConnectRequest_AlreadyRegistered|Channel management already registered|1001007|The ConnectRequest did not create a new channel since it has already been registered.|
+
+## gNMI handler error codes
+
+These are the error codes that can be returned when asking the gNMI handler in the CommunicationGateway to send something to the gRPC service.
+
+|Name|Friendly Name|Code|Message|
+|--- |--- |--- |--- |
+|GnmiHandler_SubscribeRequest_MissingSubscriptionName|gNMI handler SubscribeRequest is missing subscription name|1002001|The SubscribeRequest did not find a token in the request subject that it could use to identify the subscription with. Make sure the subject is properly formatted. The token after the SubscribeRequestCommandToken shouldn't be an empty nor a whitespace string.|
```

Comment:
```suggestion
|GnmiHandler_SubscribeRequest_MissingSubscriptionName|gNMI handler SubscribeRequest is missing subscription name|1002001|The SubscribeRequest did not find a token in the request subject that it could use to identify the subscription with. Make sure the subject is properly formatted. The token after the SubscribeRequestCommandToken should not be an empty nor a whitespace string.|
```